src/groupNoble.ts

```diff
@@ -0,0 +1,338 @@
+            case len === 1 + 2 * g.size && bytes[0] === 0x04:
+                return EltNb.deser(g, bytes)
+            default:
+                throw errDeserialization(EltNb)
+        }
+    }
+
+    static hash(g: GroupNb, msg: Uint8Array, dst: Uint8Array): EltNb {
+        const h2c = getCurve(g.id).h2c
+        const DST = new TextDecoder().decode(dst)
+        const p = h2c(msg, { DST }) as ProjPointType<bigint>
+        return new EltNb(g, p)
+    }
+}
+
+export class GroupNb implements Group {
+    static readonly Elt = EltNb
+    static readonly Scalar = ScalarNb
+    static readonly ID = GroupIDs
+    static readonly getID = getGroupID
+
+    public readonly id: GroupID
+    public readonly size: number
+
+    constructor(gid: GroupID) {
+        switch (gid) {
+            case GroupNb.ID.P256:
+                this.size = 32
+                break
+            case GroupNb.ID.P384:
+                this.size = 48
+                break
+            case GroupNb.ID.P521:
+                this.size = 66
+                break
+            default:
+                throw errBadGroup(gid)
+        }
+        this.id = gid
+    }
+
+    newScalar(): ScalarNb {
+        return ScalarNb.new(this)
+    }
+
+    newElt(): EltNb {
+        return this.identity()
+    }
+
+    identity(): EltNb {
+        return EltNb.new(this)
+    }
+
+    generator(): EltNb {
+        return EltNb.gen(this)
+    }
+
+    mulGen(s: ScalarNb): EltNb {
+        return EltNb.gen(this).mul(s)
+    }
+
+    async randomScalar(): Promise<ScalarNb> {
+        const msg = crypto.getRandomValues(new Uint8Array(this.size))
+        return ScalarNb.hash(this, msg, new Uint8Array())
+    }
+
+    async hashToGroup(msg: Uint8Array, dst: Uint8Array): Promise<EltNb> {
+        return EltNb.hash(this, msg, dst)
+    }
+
+    async hashToScalar(msg: Uint8Array, dst: Uint8Array): Promise<ScalarNb> {
+        return ScalarNb.hash(this, msg, dst)
+    }
+
+    eltDes: Deserializer<EltNb> = {
+        size: (compressed?: boolean) => {
+            return this.eltSize(compressed)
+        },
+        deserialize: (b: Uint8Array) => {
+            return this.desElt(b)
+        }
+    }
+    scalarDes: Deserializer<ScalarNb> = {
+        size: () => {
+            return this.scalarSize()
+        },
+        deserialize: (b: Uint8Array) => {
+            return this.desScalar(b)
+        }
+    }
+
+    desElt(bytes: Uint8Array): EltNb {
+        return EltNb.deserialize(this, bytes)
+    }
+
+    desScalar(bytes: Uint8Array): ScalarNb {
+        return ScalarNb.deserialize(this, bytes)
+    }
+
+    eltSize(compressed?: boolean): number {
+        return EltNb.size(this, compressed)
+    }
+
+    scalarSize(): number {
+        return ScalarNb.size(this)
+    }
+}
+
+type CurveDef = {
+    fp: Field
+    Point: ProjConstructor<bigint>
+    h2c: typeof hashToP256
+}
+const curves: Record<GroupID, CurveDef> = {
+    [GroupNb.ID.P256]: {
+        fp: FieldCons(p256.CURVE.n),
```

Comment:
```suggestion
        fp: p256.CURVE.Fp,
```

Reply:
Actually, that's Fp, the prime, not the order as is used here. I've renamed it to `fn`

Reply:
n is of course prime, but I guess it makes a a bit clearer